pyeudiw/trust/__init__.py

```diff
@@ -27,15 +29,30 @@ def __init__(self, storage: DBEngine, httpc_params, trust_anchor: str = None, **
 
     @property
     def evaluation_method(self) -> bool:
-        # TODO: implement automatic detection of trust evaluation
-        # method based on internal trust evaluetion property
-        # TODO: implement the detect of x509 trust evaluation method here
-        return self.federation
+        try:
+            getattr(self, "typ")
+        except Exception:
+            raise MissingTrustType(
+                "Unknown Trust Type: can't find 'typ' in headers"
+            )
+
+        if self.typ.upper() == "JWT":
+            return self.federation
```

Comment:
unfortunately no,
a JWT may have different typ values and be at the same time a JWT

and also a JWS json serialization would fails (even if supported by sd-jwt for example)

the trust evaluation mechanisms MUST be detected, at this stage, from the trust_chain array, if it contains JWS with entity statements OR DER bytes encoded in base64

WDYT?